examples/hosts/app-integration/external-data/src/model/taskList.ts

```diff
@@ -305,6 +305,11 @@ export class TaskList extends DataObject implements ITaskList {
 		// the "save" button case this might be fine (the user saves what they see), but in more-automatic
 		// sync'ing perhaps this should only include ack'd changes (by spinning up a second local client same
 		// as what we do for summarization).
+
+		// Force update failures to showcase retry logic.
+		if (Math.random() <= 0.33) {
+			throw new Error("Oops, something went wrong!");
+		}
```

Comment:
I think it'd be nice if this was a bit more predictable, maybe a counter that fails (and resets) every three calls. A private member variable in the class could track that. And maybe make the error "Simulated error to demonstrate a failure writing to the external service", so if someone sees it in the console they can immediately tell they can ignore it.